optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -534,27 +549,54 @@ def pre_attn_forward(
 
             softmax_mode = "fast" if flash_attention_fast_softmax else "None"
 
+            global flash_attention_in_fp8
+            if flash_attention_in_fp8 is True:
+                query_states, key_states, value_states, attention_mask = gaudi_llama_repeat_kv(
```

Comment:
consider adding 1 line comment on why this repeat kv is needed here for fp8 case.

Reply:
It tuned out not to be necessary - removed this code fragment.